### DISCUSSION TITLE
Prophet Rev2 panel

### DISCUSSION THREAD

**Discussion by synthmutt** [RomanKubiak/ctrlr#552] [General]
Is anyone working on a panel for the Prophet Rev2?

There's a Prophet 08 panel, and in the comments there were a couple questions about a similar Rev2 panel, didn't see any reply.

btw can't log in to ctrlr.org now - keep getting a Wordpress login error(?)

**Comment by synthmutt** [comment]
{ crickets }

Alright, well, I've started working on one. This way the panel will have the features I want. Given that the synth has lots of knobs and a good programming interface with its internal screen, the Ctrlr panel needs to do things that can't be done directly with the synth (or not easily done).

I didn't see that it would be easy to adapt an existing panel, so starting from a blank canvas. This one's probably a special case since it has hundreds of parameters. I've found the info needed (including program dump layout), just need to work through it (including learning Lua, and Ctrlr).

**Reply by dnaldoog** [reply]
{ crickets } is a thing it seems recently on this forum! Good luck and people are here to help once you get into the weeds (and you will).

**Reply by synthmutt** [reply]
Thanks! I think I'll post the questions I have so far in separate discussions, as they're not related just to the panel for this synth.

I have a lot of the panel working, over 100 parameters displayed. (Each Rev2 program has two layers, which are essentially entire programs on their own.) I haven't finalized the layout yet - I'm trying to show as many parameters as possible at one time on the screen, which is something that can't be done on the hardware synth - it displays one parameter at a time. Nice to see at a glance all the mod matrix slots used, e.g. to see if the mod wheel or aftertouch are assigned in the program.

I made a function to copy layers between separate programs, or just the sequence (again, something the synth front panel doesn't provide). I'm trying to figure out what librarian-type features I'd like, e.g. I'll want to offer a way to send a bank stored on disk to one of the four user banks in the Rev2 (because bank number is stored in the bank dump, so with a generic SysEx librarian the bank can only be sent back to the same one it came from).
